### PR TITLE
Problem: Request support can be improved

### PR DESCRIPTION
**Pull request by aquette**
Solution: Do it

Signed-off-by: Arnaud Quette <ArnaudQuette@Eaton.com>

### CODE REVIEW

File: src/bmsg.c [hunk 1]
```diff
@@ -603,70 +603,56 @@ int main (int argc, char *argv [])
...
 
         // Listen for the result
         zpoller_t *poller = zpoller_new (mlm_client_msgpipe (client), NULL);
         zsock_t *which = (zsock_t *) zpoller_wait (poller, timeout);
         if (which == mlm_client_msgpipe (client)) {
-            msg = mlm_client_recv (client);
+            zmsg_t *recv = mlm_client_recv (client);
+            // Check if positive
+            char *okfail = zmsg_popstr (recv);
+            if ( okfail && streq (okfail, "OK"))
```
Comment (geraldguillaume) [inline src/bmsg.c:638]: If you want a generic reply solution, you can't expect that an "OK" would belong to the reply.

**Comment by aquette** [comment]
Some example calls (should be put in the doc):
* bmsg request asset-agent ASSETS_IN_CONTAINER "GET" "" "rack controller"
bmsg - Command line tool to work with fty proto messages
I: 20-04-17 22:39:40 My address is 'bmsg.79436775'
rackcontroller-48
* bmsg request fty-sensor-gpio GPIO_MANIFEST_SUMMARY
bmsg - Command line tool to work with fty proto messages
I: 20-04-17 23:05:03 My address is 'bmsg.80960161'
DCS001
Eaton
GPOTEST
TEST
M12
Eaton
VIB001
Eaton
WLD012
Eaton
XCELW
Eaton

File: src/bmsg.c (hunk 1 above)
Comment (aquette) [reply src/bmsg.c:638]: Our protocol always answer "OK" | "OK/answer..." | "ERROR/reason" AFAICT, no?
The idea being to have a tool useful for scripting, printing the "OK" field would imply trimming it in the script (though the same apply to the disclaimer "bmsg - Command line tool to work with fty proto messages" too ;))

Comment (geraldguillaume) [reply src/bmsg.c:638]: fty-info returns first the UUID used in this request (for correlation purpose) or ERROR in case of error

**Comment by aquette** [comment]
After discussion with @geraldguillaume and his provided comments, this PR is good to merge but will be subject to some more improvements (namely, switch the ok/error logic to be more generic)

**Pull request merged**